src/main/java/nextstep/courses/domain/SessionRegisterInfo.java

```diff
@@ -6,30 +6,52 @@
 
 public class SessionRegisterInfo {
     private final SessionStatus sessionStatus;
+    private final SessionRegisteringStatus sessionRegisteringStatus;
     private final Long sessionId;
-    private final Students students;
+    private Students students;
     private final Payments payments;
 
-    public SessionRegisterInfo(Long sessionId,SessionStatus sessionStatus, Students students, Payments payments) {
+    public SessionRegisterInfo(Long sessionId,SessionStatus sessionStatus, Students students, Payments payments, SessionRegisteringStatus sessionRegisteringStatus) {
         this.sessionId = sessionId;
         this.sessionStatus = sessionStatus;
         this.students = students;
         this.payments = payments;
+        this.sessionRegisteringStatus = sessionRegisteringStatus;
     }
 
+    public void checkSessionIsOpen() {
+        if(sessionRegisteringStatus != SessionRegisteringStatus.OPEN) {
```

Comment:
객체에게 메시지를 보내서 처리해보면 어떨까요?
```suggestion
        if (sessionRegisteringStatus.isClosed()) {
```